elements/rh-secondary-nav/rh-secondary-nav-overlay.ts

```diff
@@ -0,0 +1,62 @@
+import { html, LitElement } from 'lit';
+import { customElement, state } from 'lit/decorators.js';
+
+import { ComposedEvent } from '@patternfly/pfe-core';
+import { pfelement, bound } from '@patternfly/pfe-core/decorators.js';
+
+import type { RhSecondaryNav } from './rh-secondary-nav.js';
+
+import styles from './rh-secondary-nav-overlay.css';
+
+export class SecondaryNavOverlayEvent extends ComposedEvent {
+  constructor(
+    public open: boolean,
+    public toggle: HTMLElement
+  ) {
+    super('overlay-change');
+  }
+}
+
+/**
+ * @summary A overlay to cover content when navigation is expanded.
+ */
+@customElement('rh-secondary-nav-overlay') @pfelement()
+export class RhSecondaryNavOverlay extends LitElement {
+  static readonly styles = [styles];
+
+  @state()
+  public open = false;
+
+  connectedCallback() {
+    super.connectedCallback();
+  }
+
+  render() {
+    return html``;
+  }
+
+  /**
+   * Sets or removes the hidden attribute to toggle the overlay
+   * @param toggle {HTMLElement} overlay to toggle
+   * @param state {boolean} open = true, closed = false
+   * @param parent {RhSecondaryNav} parent nav of overlay
+   */
+  @bound
+  public toggleNavOverlay(toggle: HTMLElement, state: boolean, parent: RhSecondaryNav) {
```

Comment:
rather than making this an imperative method call, why not just set `.open = true`, then handle the side effects here (or even in css)
```ts
@property({ type: Boolean, reflect: true }) open = false;
```
```css
:host(:not([open])) {
  display: none;
}
```

Reply:
Fixed in a4fa730f5058f56e4e829a9938b5264bf25a2b8d